packages/react-router-dom/index.tsx

```diff
@@ -339,11 +324,11 @@ export function BrowserRouter({
   });
   let setState = React.useCallback(
     (newState: { action: NavigationType; location: Location }) => {
-      startTransitionImpl
+      future && future.v7_startTransition && startTransitionImpl
         ? startTransitionImpl(() => setStateImpl(newState))
         : setStateImpl(newState);
     },
-    [setStateImpl]
+    [setStateImpl, future]
```

Comment:
We probably want to destructure "v7_startTransition" off of "future" as we don't ever want that object identity to accidentally change.